### PR TITLE
Fix and simplify attachment dimension mismatch check

### PR DESCRIPTION
**Pull request by DasEtwas**
This also improves error messages with detailed dimension and attachment type name info which developers can hopefully recognize.

**Connections**
None

**Description**
Before this commit, depth/stencil attachments are not considered when evaluating the renderpasses' dimension, which will result in a "NoAttachment" error with a confusing display message.

**Testing**
Tested by running an application making use of color-only and depth/stencil-only attachment renderpasses.

Tested by running boids, hello-triangle and shadow examples.

### CODE REVIEW

File: wgpu-core/src/command/render.rs [hunk 1]
```diff
@@ -387,10 +387,22 @@ pub enum RenderPassErrorInner {
     Encoder(#[from] CommandEncoderError),
     #[error("attachment texture view {0:?} is invalid")]
     InvalidAttachment(id::TextureViewId),
-    #[error("there are attachments")]
-    NoAttachments,
-    #[error("attachments have different sizes")]
-    MismatchAttachments,
+    #[error("necessary attachments are missing")]
+    MissingAttachments,
+    #[error("color and/or depth attachments have differing sizes: a {mismatching_attachment_type_name} attachment of size {mismatching_dimensions_width}x{mismatching_dimensions_height} is not compatible with a {previous_attachment_type_name} attachment's size ({previous_dimensions_height}x{previous_dimensions_width})",
```
Comment (kvark) [inline wgpu-core/src/command/render.rs:392]: just printing the tuple in the error would be nearly as clear, and would save us a few keystrokes here. Perhaps, we can avoid those `previous_dimensions_width = ...` defines, and just print the whole tuples?

File: wgpu-core/src/command/render.rs [hunk 2]
```diff
@@ -526,9 +538,12 @@ impl<'a, B: GfxBackend> RenderPassInfo<'a, B> {
 
         let mut render_attachments = AttachmentDataVec::<RenderAttachment>::new();
 
+        // need to be initialized to None!, check `AttachmentsDimensionMismatch` error
         let mut attachment_width = None;
         let mut attachment_height = None;
-        let mut valid_attachment = true;
+        let mut attachment_type_name = None;
```
Comment (kvark) [inline wgpu-core/src/command/render.rs:544]: could we just initialize it to `""`? the option isn't helping here, I think

**Review by kvark — COMMENTED** [review]
Very nice! just a few small suggestions here

**Comment by DasEtwas** [comment]
Applied the suggested changes

**Review by kvark — APPROVED** [review]
Thank you!
bors r+

**Comment by kvark** [comment]
Note: I'm refactoring this code a bit in https://github.com/gfx-rs/wgpu/pull/1147/commits/a4f897b5d78268cfab1250a24088415baf69dad1

**Pull request merged**